File: java/src/test/java/com/kcl/ParseFileTest.java
package com.kcl;

import java.nio.file.Path;
import java.nio.file.Paths;

import org.junit.Assert;
import org.junit.Test;

import com.kcl.api.API;
import com.kcl.api.Spec.ParseFileArgs;
import com.kcl.api.Spec.ParseFileResult;

public class ParseFileTest {
    @Test
    public void testParseFile() throws Exception {
        Path path = Paths.get("src", "test_data", "parse", "main.k");
        ParseFileArgs args = ParseFileArgs.newBuilder().setPath(path.toString()).build();

        API apiInstance = new API();
        ParseFileResult result = apiInstance.parseFile(args);
        Assert.assertNotNull(result.getAstJson());
        Assert.assertEquals(result.getDepsCount(), 2);
    }
}

File: kotlin/src/test/kotlin/com/kcl/ParseFileTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.parseFileArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals
import java.nio.file.Paths

class ParseFileTest {
    companion object {
        private val TEST_FILE = Paths.get("src", "test_data", "parse", "main.k").toString()
    }

    @Test
    fun testParseFileApi() {
        val args = parseFileArgs { path = TEST_FILE }
        val api = API()
        val result = api.parseFile(args)
        assertEquals(result.depsCount, 2)
    }
}
